### DISCUSSION TITLE
Using in a kernel context

### DISCUSSION THREAD

**Discussion by lretq** [spaskalev/buddy_alloc#43] [General]
Is there a way to bypass e.g. __subsf3 undefined symbol (with -ffreestanding) errors?

I want to use this for memory management in a kernel.

**Comment by spaskalev** [comment]
Can you paste the complete command and output that you are using ? I have some other (private) users in a kernel and what they have done was to comment out (via a define guard) the debug functions due to having no FILE types in their kernel.

The basic allocator doesn't use any IO so you should be good to just remove these functions - or fork and re-implement them so that they can work with your kernel.

**Reply by lretq** [reply]
I already did that, as I don't have any FILE structures etc..

The problem is float/math related (I think?)

For example this:
```
ld.lld: error: undefined symbol: __floatundisf
>>> referenced by buddy_alloc.h:1715 (sources/buddy/buddy_alloc.h:1715)
```

**Reply by lretq** [reply]
I'm missing these compiler builtins. I am asking if the float related things have to be float?

**Reply by spaskalev** [reply]
You can remove the float functions too - they are only used in the fragmentation calculation, which is a utility metric and not required for allocation.

**Reply by lretq** [reply]
Alright, thanks!
